sst/SimEngCoreWrapper.cc

```diff
@@ -100,12 +109,14 @@ void SimEngCoreWrapper::init(unsigned int phase) {
 bool SimEngCoreWrapper::clockTick(SST::Cycle_t current_cycle) {
   // Tick the core and memory interfaces until the program has halted
   if (!core_->hasHalted() || dataMemory_->hasPendingRequests()) {
+    // Ticking dataMemory just increases cycle count. Tick first to ensure
```

Comment:
This was intentional, as It changes the format from main, in main `data memory->tick()` is after `core->tick()`. Since the core ticks before data memory in SSTSimeng, the cycle count inside SimEngMemInterface isn't accurate as a result clock cycle calculations are off by 1.

Reply:
Format as in comment format, so it could just be:
```
    // Tick data memory
    dataMemory_->tick();

    // Tick the core
    core_->tick();

    // Tick instruction memory
    instructionMemory_->tick();
```

Reply:
All information regarding the body of `tick()` should be coupled with its declaration/definition (but none needed as it's self-explanatory)

Reply:
I have corrected this, please check